redis/src/cluster_async/mod.rs

```diff
@@ -988,35 +989,50 @@ where
         ) -> (
-            Vec<(ArcStr, Receiver<Result<Response, RedisError>>)>,
-            Vec<PendingRequest<C>>,
+            Vec<(Option<ArcStr>, Receiver<Result<Response, RedisError>>)>,
+            Vec<Option<PendingRequest<C>>>,
         ) {
             iterator
-                .filter_map(|(cmd, (identifier, conn))| {
+                .map(|tuple_opt| {
                     let (sender, receiver) = oneshot::channel();
-                    connections_container
-                            .address_for_identifier(&identifier) // TODO - this lookup can be avoided, since address is only needed on Special aggregates.
-                            .map(|address| {
-                                (
-                                    (address, receiver),
-                                    PendingRequest {
-                                        retry: 0,
-                                        sender,
-                                        info: RequestInfo {
-                                            cmd: CmdArg::Cmd {
-                                                cmd,
-                                                routing: InternalSingleNodeRouting::Connection {
-                                                    identifier,
-                                                    conn,
-                                                }.into(),
-                                            },
-                                        },
+                    if let Some((cmd, identifier, conn, address)) =
+                        tuple_opt.and_then(|(cmd, (identifier, conn))| {
+                            // TODO - this lookup can be avoided, since address is only needed on Special aggregates.
+                            connections_container
+                                .address_for_identifier(&identifier)
+                                .map(|address| (cmd, identifier, conn, address))
+                        })
+                    {
+                        (
+                            (Some(address), receiver),
+                            Some(PendingRequest {
+                                retry: 0,
+                                sender,
+                                info: RequestInfo {
+                                    cmd: CmdArg::Cmd {
+                                        cmd,
+                                        routing: InternalSingleNodeRouting::Connection {
+                                            identifier,
+                                            conn,
+                                        }
+                                        .into(),
                                     },
-                                )
-                            })
+                                },
+                            }),
+                        )
+                    } else {
+                        let _ = sender.send(Err((
```

Comment:
If I understand it right, if we have one cmd that we can't find the connection for it, all of the multi-command will be failed. right?
If so, why do we need to continue mapping after we send the error?

Reply:
This function doesn't take the `response_policy` into account, in order to remain simple. You're right that for some response policies we could fail immediately, but that would require more code complexity, and it would mean that if a function has side effects (SCRIPT LOAD, for example), no node will be affected - it would be equivalent to all nodes failing.